### PR TITLE
asebadummynode can request ephemeral ports

### PR DESCRIPTION
**Pull request by davidjsherman**
First part of issue #544: asebadummynode now can ask for an ephemeral port using the `--port 0` option. Additionally, the `--basePort` option allows one to start numbering anywhere, not just 33333.
```
% asebadummynode -p 0
tcp:port=59048
% asebadummynode -p 20000 1
tcp:port=20000
% asebadummynode -b 20000 2
tcp:port=20002
% asebadummynode
tcp:port=33333
```
This PR depends on Dashel >= 1.2.0 in order to correctly print the target with the generated port number; otherwise, it is reported as `tcp:port=0`.

```
Usage: asebadummynode [--basePort|-b BASE, default 33333] [ID, from 0 to 9]
Usage: asebadummynode --port|-p PORT [ID, from 0 to 9]
Usage: asebadummynode --help|-h
Creates one node dummynode-ID with node id ID+1 listening on port:
 - BASE+ID, if BASE != 0 and BASE+ID is available
 - PORT, if PORT != 0 and PORT is available
 - a dynamically chosen port, if PORT or BASE == 0.
The Dashel target is printed on stdout.
```

### CODE REVIEW

**Comment by stephanemagnenat** [comment]
Thank you for these features. I am wondering if BASE is necessary. Why not simply `-p PORT [ID]`?

**Comment by davidjsherman** [comment]
If we are certain that the legacy auto-increment behavior is only to avoid port conflicts, then there is no particular reason. I don't know whether there is a scenario in use out there where predictable port numbers for sequential node ids are assumed.

I admit that there is not much difference between the two lines
```
for i in 0 1 2 3 4 5 6 7 8; do asebadummynode -p $[ 20000 + $i ] $i & done
for i in 0 1 2 3 4 5 6 7 8; do asebadummynode -p 20000 $i & done
```

**Comment by davidjsherman** [comment]
Getting the targets without Zeroconf is not easy because one needs to read stdout of what is typically a backgrounded process. One way to do it is:
```
targets=$( for i in 0 1 2; do exec 3< <(asebadummynode -p 0 $i); read <&3; echo $REPLY; done; )
```
using [process substitution](https://www.gnu.org/software/bash/manual/bash.html#Process-Substitution) in Bash.

**Comment by stephanemagnenat** [comment]
The legacy auto-increment behavior is currently only to avoid port conflicts, as the new switch does automatically remap conflicting ids. The original switch did not, so auto-incrementing the node ID was a good solution. Therefore, I suggest to keep it simple and only use -p PORT [ID] as command-line argument.

I think that either setting port by hand, or using zeroconf, or using process substitution covers a large enough range of cases.

File: targets/dummy/dummynode.cpp [hunk 1]
```diff
@@ -82,29 +82,32 @@ class AsebaNode: public Dashel::Hub
 		vm.variablesSize = sizeof(variables) / sizeof(sint16);
 	}
 	
-	void listen(int basePort, int deltaPort)
+	Dashel::Stream* listen(const int port, int deltaNodeId)
```
Comment (stephanemagnenat) [inline targets/dummy/dummynode.cpp]: Why port is const and not deltaNodeId?

**Comment by stephanemagnenat** [comment]
Perfect, let's merge, thank you :-)

**Pull request merged**